contracts/allocators/NFTXAllocator.sol

```diff
@@ -0,0 +1,254 @@
+    }
+
+
+    /**
+     * @notice claims rewards from the vault.
+     */
+
+    function harvestAll(address _token) override external onlyPolicy {
+        // We only want to allow harvesting from a specified liquidity pool mapping
+        require(stakingTokenInfo[_token].exists, "Unsupported token");
+        require(stakingTokenInfo[_token].isLiquidityPool, "Must be liquidity staking token");
+
+        // Send a request to the treasury to claim rewards from the NFTX liquidity staking pool
+        treasury.claimNFTXRewards(
+            address(liquidityStaking),
+            stakingTokenInfo[_token].vaultId,
+            stakingTokenInfo[_token].rewardToken
+        );
+    }
+
+
+    /**
+     * @notice sends any ERC20 token in the contract to caller.
+     */
+
+    function rescue(address _token) external override onlyGovernor {
+        // If the token is known, then we shouldn't be able to rescue it
+        require(!stakingTokenInfo[_token].exists, "Known token cannot be rescued");
+
+        // Get the amount of token held on contract
+        uint256 _amount = IERC20(_token).balanceOf(address(this));
+
+        // Confirm that we hold some of the specified token
+        require(_amount > 0, "Token not held in contract");
+
+        // Send to Governor
+        IERC20(_token).safeTransfer(msg.sender, _amount);
+    }
+
+
+    /**
+     * @notice withdraws asset from treasury, deposits asset into NFTX staking.
+     */
+
+    function deposit(address _token, uint256 _amount) external override onlyPolicy {
+        require(stakingTokenInfo[_token].exists, "Unsupported staking token");
+        require(dividendTokenInfo[_token].underlying == _token, "Unsupported dividend token");
+
+        // Ensure that a calculator exists for the `dividendTokenInfo[_token].xToken`
+        require(treasury.bondCalculator(dividendTokenInfo[_token].xToken) != address(0), "Unsupported xToken calculator");
+
+        // Retrieve amount of asset from treasury, decreasing total reserves
+        treasury.allocatorManage(_token, _amount);
+
+        uint256 value = treasury.tokenValue(_token, _amount);
+        emit TreasuryAssetDeployed(_token, _amount, value);
+
+        // Approve and deposit into inventory pool, returning xToken
+        if (stakingTokenInfo[_token].isLiquidityPool) {
+            IERC20(_token).approve(address(liquidityStaking), _amount);
+            liquidityStaking.deposit(stakingTokenInfo[_token].vaultId, _amount);
+        } else {
+            IERC20(_token).approve(address(inventoryStaking), _amount);
+            inventoryStaking.deposit(stakingTokenInfo[_token].vaultId, _amount);
+        }
+    }
+
+    /**
+     * @notice Withdraws from staking pool, and deposits asset into treasury.
+     */
+
+    function withdraw(address _token, uint256 _amount) external override onlyPolicy {
+        require(stakingTokenInfo[_token].exists, "Unsupported staking token");
```

Comment:
Since `stakingTokenInfo` is being accessed a lot as a storage variable, might be a good idea to save some gas by storing in a temp variable first before using it